hashintel/hash
# Fetch `authenticatedUser` server-side in Next.js

## 🌟 What is the purpose of this PR?




This PR adds `getInitialProps` to `_app.page.ts` which fetches the `authenticatedUser` server-side on _every_ page-load in the HASH App. This allows the `authenticatedUser` to be available on first-render of the application (when the user is logged in).

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1203363157432081/1203451531168818/f) _(internal)_

## 🔍 What does this change?




- replaces the existing `useAuthenticatedUser` with a simplified version of the hook, that makes use of react context to minimise data fetching
- fixes some routing logic to allow pages to be publicly accessible in readonly mode
- updates the `oryKratosClient` to be work on both the server and the client

## 📜 Does this require a change to the docs?



Not that I'm aware of.

## ⚠️ Known issues



## 🐾 Next steps



## 🛡 What tests cover this?



None, playwright is not working. The app was tested manually.

## ❓ How to test this?



1. Run the HASH App, and login.
2. Ensure the app is working as previously, and that redirects are happening before the page is rendered.

## 📹 Demo
